core/gui/src/app/workspace/component/result-panel/result-panel.component.ts

```diff
@@ -281,14 +287,33 @@ export class ResultPanelComponent implements OnInit, OnDestroy {
 
   resetPanelPosition() {
     this.dragPosition = { x: this.returnPosition.x, y: this.returnPosition.y };
+    this.isDocked = true;
+  }
+
+  checkPanelIsDocked() {
+    this.isDocked = this.returnPosition.x === this.dragPosition.x && this.returnPosition.y === this.dragPosition.y;
+  }
+
+  getResizeDirections() {
+    return (this.isDocked ? this.dockedResizeDirections : this.undockedResizeDirections) as NzResizeDirection[];
+  }
+
+  handleStartDrag() {
+    this.isDocked = false;
   }
 
   onResize({ width, height }: NzResizeEvent) {
     cancelAnimationFrame(this.id);
+    this.updateReturnPosition(this.height, height);
     this.id = requestAnimationFrame(() => {
       this.width = width!;
       this.height = height!;
       this.resizeService.changePanelSize(this.width, this.height);
     });
   }
+
+  updateReturnPosition(prevHeight: number, newHeight: number | undefined) {
+    if (!isDefined(newHeight)) return;
```

Comment:
have some comments on top of this function to explain the reason why you are updating the returnPosition

Reply:
added comment

Reply:
You can make it shorter. Something like "This return position update ensures that even if the panel gets resized, it can be docked correctly to the left-bottom corner of the canvas. "

Reply:
updated comment